src/plugins/intel_npu/src/plugin/npuw/just_sync_infer_request.cpp

```diff
@@ -452,14 +479,13 @@ void ov::npuw::JustInferRequest::unpack_closure(std::size_t idx, RqPtr request)
 
         const auto closure_param_id = comp_model_desc.param_base + cidx;
         auto& iport = func_desc.compiled_model->inputs()[closure_param_id];
-        auto clparam = request->get_tensor(iport);
-        if (closure.get_element_type() != clparam->get_element_type()) {
+        if (comp_model_desc.unpack_required[cidx]) {
             // Remember where the unpack is required
             closure_unpack_required.push_back(cidx);
-        } else {
+        } else if (comp_model_desc.update_required[cidx]) {
             // Easy case, just set one to another. Copy_to is also possible
             // and even may be preferrable for some devices, like this:
-            // ```ov::get_tensor_impl(closure)->copy_to(clparam._ptr);'''
+            // ```ov::get_tensor_impl(closure)->copy_to(request->get_tensor(iport)._ptr);'''
             request->set_tensor(iport, ov::get_tensor_impl(closure));
         }
```

Comment:
This is the right logic. If update is required, `set_tensor()` is called. But you initialize this flag based on inverse of this logic

Reply:
Done